runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -366,28 +383,13 @@ impl<T: Config> Pallet<T> {
 				)
 			};
 
-			// In case of an overweight block, consume up to the entire block weight
-			// in disputes, since we will never process anything else, but invalidate
-			// the block. It's still reasonable to protect against a massive amount of disputes.
-			if candidates_weight
-				.saturating_add(bitfields_weight)
-				.saturating_add(disputes_weight)
-				.any_gt(max_block_weight)
-			{
-				log::warn!("Overweight para inherent data reached the runtime {:?}", parent_hash);
-				backed_candidates.clear();
-				candidates_weight = Weight::zero();
-				signed_bitfields.clear();
-				bitfields_weight = Weight::zero();
-			}
```

Comment:
I don't understand this:

1. Any change should also have a corresponding change in create_inherent.
2. Why would we want to only limit disputes? If we want to limit weight more aggressively we should limit it for all of paras_inherent.
3. Why was this clearing dropped?

Goal is to not have overweight blocks, but still disputes are the most important thing to include. In particular not including candidates when there are lot's of disputes is something we absolutely want. As each included candidate is a potential future dispute.

Reply:
First we do get some back pressure which helps handling the dispute storm, at the same time we will be processing (and slashing) at max possible speed, which is good in general to avoid issues like validators avoiding getting slashed by causing a system overload and also again helps with mitigating the situation, because once we have disabling, having those validators disabled quickly will restore the system health quicker.

In general, a disputes should be absolutely exceptional, a dispute storm even more so, but if it happens, resolving those disputes is the most important thing the system is supposed to accomplish anything else is secondary. Parachain block times are supposed to suffer on disputes, that's why we want validators on the wrong side to pay for them.

Reducing dispute processing by xx%, but processing candidates at full speed is the [exact opposite of what we want](https://github.com/paritytech/polkadot/issues/5898).

Reply:
Put differently: It must never ever happen that we cannot keep up with disputes. Reducing our dispute processing capabilities while assigning those freed resources to the production of even more disputes is a recipe for exactly that.

Reply:
Totally agree with `resolving those disputes is the most important thing the system`, but they get resolved off-chain. AFAIU we only put dispute votes on chain to enable slashing. I agree that fast slashing/disabling is something we want to happen fast. But in all practical cases, it will mean that the validator gets a little more time, depending on when we actually slash/disable.

Reply:
> Put differently: It must never ever happen that we cannot keep up with disputes. Reducing our dispute processing capabilities while assigning those freed resources to the production of even more disputes is a recipe for exactly that.

I think the right back pressure mechanism for stopping disputes is slashing/disable, not us slowing down the whole ecosystem but I get your point. You are assuming that without this backpressure mechanism the system would allow for a scenario as in https://github.com/paritytech/polkadot/issues/5898 . IMO for an attacker to be able to back and approve a candidate because of disputes high load is betting on secondary checkers to never be able to issue disputes on a specific block. I can only speculate if such an attack is possible in practice if we have slashing/disable enabled.

As far as I have seen, there is enough backpressure from other parts of the system like slow availability or finality lag that have the same effect. So I'd rely on those instead of this early one which gives false positives in the sense that it slows down all parachains when actually the load is manageble.

Reply:
If we have slashing/disabling in place then this also is not an issue. If the load is manageable, then we will also barely filter out candidates. You have to check the create_inherent function, there we fill up the block as much as possible. If there space left after disputes, we put in bitfields, then backed candidates. That code in `enter_inner` which just clears out everything other than disputes unconditionally should actually never run if `create_inherent` is working correctly.

And we have still have seen occasions where we had trouble keeping up with disputes, so cutting down a back pressure mechanism that is already there is just not good.

Reply:
after some more thinking and considering your points https://github.com/paritytech/polkadot/issues/6919#issuecomment-1480017105 should be something reasonable.